ballerina/client.bal

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Consists of APIs to integrate with LDAP.
+public isolated client class Client {
+
+    # Gets invoked to initialize the LDAP client.
+    #
+    # + config - The configurations to be used when initializing the client
+    # + return - An error if client initialization failed
+    public isolated function init(*ConnectionConfig config) returns error? {
+        self.generateLdapClient(config);
+    }
+
+    private isolated function generateLdapClient(ConnectionConfig config) = @java:Method {
+        'class: "io.ballerina.lib.ldap.Ldap"
+    } external;
```

Comment:
How about

```suggestion
    private isolated function initLdapConnection(ConnectionConfig config) = @java:Method {
        'class: "io.ballerina.lib.ldap.Ldap"
    } external;
```
?

IMO here we are not "generating" something, rather initialisialising a native level client/connection, right?

Reply:
Yeah. Fixed it